bin/middlewares/query_bbox.js

```diff
@@ -0,0 +1,55 @@
+const axios = require('axios');
+
+module.exports = function (config) {
+  const geocoderUrl = config.services.geocoder.url;
+  const idunnTimeout = Number(config.server.services.idunn.timeout);
+  if (isNaN(idunnTimeout)) {
+    throw new Error(
+      `Invalid config: idunn timeout is set to "${config.server.services.idunn.timeout}"`
+    );
+  }
+
+  async function getFullTextBbox(query, langCode) {
+    try {
+      const response = await axios.get(geocoderUrl, {
+        params: {
+          lang: langCode,
+          nlu: 'true',
+          q: query,
+        },
+        timeout: idunnTimeout,
+      });
+      const intention = (response.data.intentions || [])[0];
+      if (intention && intention.filter && intention.filter.bbox) {
+        return intention.filter.bbox;
+      }
+      return null;
+    } catch (error) {
+      if (error.response && error.response.status === 404) {
+        return null;
+      }
+      throw error;
+    }
+  }
+
+  return function (req, res, next) {
+    const absoluteUrl = `${req.protocol}://${req.get('host')}${req.originalUrl}`;
+    const fullTextQuery = new URL(absoluteUrl).searchParams.get('q');
```

Comment:
You would probably prefer using [`req.query`](https://expressjs.com/en/api.html#req.query)
```suggestion
    const fullTextQuery = req.query.q;
```

Reply:
Ah yes, I'm so used to code on the client that I forgot Express has some easy shortcuts :)
Thanks!